### ISSUE TITLE
"Failed to bump fs.file-max, ignoring: Invalid argument" after kernel upgraded from 5.1.9 to 5.1.10

### ISSUE THREAD

**Issue by enihcam** [systemd/systemd#12803]
**systemd version the issue has been seen with**
> 242.29

**Used distribution**
 > Archlinux

**Expected behaviour you didn't see**
 > fs.file-max bumped.

**Unexpected behaviour you saw**
 > "Failed to bump fs.file-max, ignoring: Invalid argument" in dmesg

**Steps to reproduce the problem**
upgrade kernel from 5.1.9 to 5.1.10.

kernel config: [config.gz](https://github.com/systemd/systemd/files/3293778/config.gz)

**Comment by Tatsh** [comment]
Confirming same issue on Gentoo.

**Comment by zdzichu** [comment]
5.1.10 contains this patch: sysctl: return -EINVAL if val violates minmax
https://lkml.org/lkml/2019/6/1/307

Seem related.

**Comment by yuwata** [comment]
So, at least, this is not our bug, as we just write configured values. Can I close this?

**Comment by enihcam** [comment]
It looks like systemd tries to write ULONG_MAX into fs.file-max.
```
#if BUMP_PROC_SYS_FS_FILE_MAX
        /* I so wanted to use STRINGIFY(ULONG_MAX) here, but alas we can't as glibc/gcc define that as
         * "(0x7fffffffffffffffL * 2UL + 1UL)". Seriously. 😢 */
        if (asprintf(&t, "%lu\n", ULONG_MAX) < 0) {
                log_oom();
                return;
        }

        r = sysctl_write("fs/file-max", t);
        if (r < 0)
                log_full_errno(IN_SET(r, -EROFS, -EPERM, -EACCES) ? LOG_DEBUG : LOG_WARNING, r, "Failed to bump fs.file-max, ignoring: %m");
#endif
```

**Comment by poettering** [comment]
Fix waiting in #12808
